web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/ManualHelmDialog.tsx

```diff
@@ -19,24 +19,98 @@
 import Dialog, { DialogContent, DialogFooter } from 'design/DialogConfirmation';
 import { Box, Flex, ButtonPrimary, ButtonSecondary, Text } from 'design';
 
-import React from 'react';
+import React, { Suspense } from 'react';
 
 import styled from 'styled-components';
 
+import { Spinner } from 'design/Icon';
+
+import * as Icons from 'design/Icon';
+
 import { TextSelectCopyMulti } from 'teleport/components/TextSelectCopy';
 import { CommandBox } from 'teleport/Discover/Shared/CommandBox';
 
+import { useJoinTokenSuspender } from 'teleport/Discover/Shared/useJoinTokenSuspender';
+import { ResourceKind, TextIcon } from 'teleport/Discover/Shared';
+import { JoinToken } from 'teleport/services/joinToken';
+import { CatchError } from 'teleport/components/CatchError';
+
 type ManualHelmDialogProps = {
-  command: string;
+  setJoinTokenAndGetCommand(token: JoinToken): string;
   confirmedCommands(): void;
   cancel(): void;
 };
 
+export default function Container(props: ManualHelmDialogProps) {
+  return (
+    <CatchError
+      fallbackFn={fallbackProps => (
+        <DummyDialog error={fallbackProps.error} cancel={props.cancel} />
+      )}
+    >
+      <Suspense
+        fallback={<DummyDialog showSpinner={true} cancel={props.cancel} />}
+      >
+        <ManualHelmDialog {...props} />
+      </Suspense>
+    </CatchError>
+  );
+}
+
+type DummyDialogProps = {
+  cancel: () => void;
+  error?: Error;
+  showSpinner?: boolean;
+};
+
+const DummyDialog = ({ error, cancel, showSpinner }: DummyDialogProps) => {
+  return (
+    <Dialog onClose={cancel} open={true}>
+      <DialogContent width="850px" mb={0}>
+        <Text bold caps mb={4}>
+          Manual EKS Cluster Enrollment
+        </Text>
+        {showSpinner && (
+          <Flex mb={4} justifyContent="center">
+            <Spin>
+              <Spinner />
+            </Spin>
+          </Flex>
+        )}
+        {error && (
+          <Box mb={4}>
+            <TextIcon mt={3}>
+              <Icons.Warning size="medium" ml={1} mr={2} color="error.main" />
+              Encountered an error: {error.message}
+            </TextIcon>
+          </Box>
+        )}
+      </DialogContent>
+      <DialogFooter alignItems="center" as={Flex} gap={4}>
+        <ButtonPrimary width="50%" onClick={() => {}} disabled>
+          I ran these commands
+        </ButtonPrimary>
+        <ButtonSecondary width="50%" onClick={cancel}>
+          Cancel
+        </ButtonSecondary>
+      </DialogFooter>
+    </Dialog>
+  );
+};
+
 export function ManualHelmDialog({
-  command,
+  setJoinTokenAndGetCommand,
   cancel,
   confirmedCommands,
 }: ManualHelmDialogProps) {
+  const { joinToken } = useJoinTokenSuspender([
+    ResourceKind.Kubernetes,
+    ResourceKind.Application,
+    ResourceKind.Discovery,
+  ]);
+
+  const command = setJoinTokenAndGetCommand(joinToken);
```

Comment:
my knee jerk response is to wrap this in useEffect since `setJoinTokenAndGetCommand` calls a react setter (worst case this could result in infinite re-render), i'm betting this is why you get this

<img width="1379" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/7bdb1af7-1bab-4537-a52a-9f1199dcb740">

Reply:
Applied 👍